Arrowgene.Ddon.GameServer/Characters/ItemManager.cs

```diff
@@ -585,8 +589,40 @@ public uint LookupItemByUID(DdonServer<GameClient> server, string itemUID)
 
             return item.ItemId;
         }
+
+
+        public List<CDataItemUpdateResult> UpgradeStorageItem(DdonGameServer server, GameClient client, UInt32 characterID, StorageType storageType, Item newItem, byte slotNo)
+        {
+            client.Character.Storage.GetStorage(storageType).SetItem(newItem, 1, slotNo);
+            server.Database.UpdateStorageItem(characterID, storageType, slotNo, 1, newItem);
+            
+            CDataItemUpdateResult updateResult = new CDataItemUpdateResult();
+            updateResult.ItemList.ItemUId = newItem.UId;
+            updateResult.ItemList.ItemId = newItem.ItemId;
+            updateResult.ItemList.ItemNum = 1;
+            updateResult.ItemList.Unk3 = newItem.Unk3;
+            updateResult.ItemList.StorageType = storageType;
+            updateResult.ItemList.SlotNo = slotNo;
+            updateResult.ItemList.Color = newItem.Color;
+            updateResult.ItemList.PlusValue = newItem.PlusValue;
+            updateResult.ItemList.Bind = false;
+            updateResult.ItemList.EquipPoint = newItem.EquipPoints;
+            updateResult.ItemList.EquipCharacterID = 0;
+            updateResult.ItemList.EquipPawnID = 0;
+            updateResult.ItemList.WeaponCrestDataList = newItem.WeaponCrestDataList;
+            updateResult.ItemList.AddStatusData = newItem.AddStatusParamList;
+            updateResult.ItemList.EquipElementParamList = newItem.EquipElementParamList;
+            updateResult.UpdateItemNum = 1;
+
+            Logger.Debug($"Upgraded {newItem.UId} Item in DataBase");
+
+            return new List<CDataItemUpdateResult> { updateResult };
```

Comment:
Why do we return a list if this function always returns a list of 1?

Reply:
It makes for consistency with other ItemManager functions, i think this isn't the first ItemManager function that always returns a list of one item